### PR TITLE
Add localization selection

### PR DESCRIPTION
**Pull request by roman-drozd-it**
A language is detected when the application is opened for the first time and it can be changed in **TopBar** and **Settings**. 
The language selection is persistent (redux-persist) until changed by the user.
Configuration is now in _app/src/translations/locales.js_.

### CODE REVIEW

File: app/src/components/Controls/TopBar.js [hunk 1]
```diff
@@ -840,7 +871,10 @@ TopBar.propTypes =
 	canLock              : PropTypes.bool.isRequired,
 	canPromote           : PropTypes.bool.isRequired,
 	classes              : PropTypes.object.isRequired,
-	theme                : PropTypes.object.isRequired
+	theme                : PropTypes.object.isRequired,
```
Comment (havfo) [inline app/src/components/Controls/TopBar.js:874]: Fix indentation

File: app/src/components/Controls/TopBar.js [hunk 2]
```diff
@@ -933,7 +969,9 @@ export default withRoomContext(connect(
 				prev.me.roles === next.me.roles &&
 				prev.toolarea.unreadMessages === next.toolarea.unreadMessages &&
 				prev.toolarea.unreadFiles === next.toolarea.unreadFiles &&
-				prev.toolarea.toolAreaOpen === next.toolarea.toolAreaOpen
+				prev.toolarea.toolAreaOpen === next.toolarea.toolAreaOpen &&
+				prev.locale === next.locale &&
```
Comment (havfo) [inline app/src/components/Controls/TopBar.js:973]: Needs to be `prev.intl.locale`, this refers to the global store.

File: app/src/components/Controls/TopBar.js [hunk 3]
```diff
@@ -933,7 +969,9 @@ export default withRoomContext(connect(
...
+				prev.localesList === next.localesList
```
Comment (havfo) [inline app/src/components/Controls/TopBar.js:974]: Same as previous

File: app/src/components/Settings/AppearenceSettings.js [hunk 4]
```diff
@@ -229,8 +278,10 @@ export default connect(
 			return (
 				prev.me.browser === next.me.browser &&
 				prev.room === next.room &&
-				prev.settings === next.settings
+				prev.settings === next.settings &&
+				prev.locale === next.locale &&
```
Comment (havfo) [inline app/src/components/Settings/AppearenceSettings.js:282]: Should be `prev.intl.locale`

File: app/src/translations/locales.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	{ name: 'Croatian',				file: 'hr', locale: [ 'hr', 'hr-hr' ] },
+	{ name: 'Danish',				file: 'dk', locale: [ 'dk', 'dk-dk' ] },
+	{ name: 'French',				file: 'fr', locale: [ 'fr', 'fr-fr' ] },
+	{ name: 'German',				file: 'de', locale: [ 'de', 'de-de' ] },
+	{ name: 'Greek',					file: 'el', locale: [ 'el', 'el-el' ] },
+	{ name: 'Hindi',					file: 'hi', locale: [ 'hi', 'hi-hi' ] },
+	{ name: 'Hungarian',				file: 'hu', locale: [ 'hu', 'hu-hu' ] },
+	{ name: 'Italian',				file: 'it', locale: [ 'it', 'it-it' ] },
+	{ name: 'Latvian',				file: 'lv', locale: [ 'lv', 'lv-lv' ] },
+	{ name: 'Norwegian',				file: 'nb', locale: [ 'nb', 'nb-nb' ] },
```
Comment (havfo) [inline app/src/translations/locales.js:15]: If using region code, it is nb-no

**Comment by roman-drozd-it** [comment]
Done.

**Comment by roman-drozd-it** [comment]
I've resolved the conflicts, cause It's been a while when that pull request was added. If no objections, can we merge it to develop?
I saw some people asked about that feature.

**Pull request merged**